src/mavlink-router/endpoint.cpp

```diff
@@ -191,11 +191,11 @@ int Endpoint::read_msg(struct buffer *pbuf, int *target_sysid, int *target_compi
             return 0;
 
         msg_id = hdr->msgid;
-        payload = rx_buf.data + sizeof(*hdr);
+        pbuf->payload = payload = rx_buf.data + sizeof(*hdr);
         seq = hdr->seq;
-        *src_sysid = hdr->sysid;
-        *src_compid = hdr->compid;
-        payload_len = hdr->payload_len;
+        pbuf->src_sysid = *src_sysid = hdr->sysid;
```

Comment:
I did not checked if you are doing it in the further patches(as I will review the other patches other I want to leave this comment here) but with this fields in buffer we could remove then from read_msg() and make this:
pbuf->src_sysid = *src_sysid = hdr->sysid; to this: pbuf->src_sysid = hdr->sysid;

Reply:
As you noticed, yes I'm doing in a following patch, otherwise there would be a single giant patch that's hard to review and validate